File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
private:
    void findSubset(int index, vector<int> &temp, vector<vector<int>> &result,vector<int>nums ){
        result.push_back(temp);
        if(index==nums.size()) return;

        for(int i=index ; i<nums.size(); i++){
            if(i > index && nums[i] == nums[i-1]) continue; //skipping dup
            temp.push_back(nums[i]);
            findSubset(i+1,temp,result,nums);
            temp.pop_back();
        }
    }
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int> temp;
        vector<vector<int>>result;
        findSubset(0,temp,result,nums);
        return result;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    private void findSubset(int index , List<List<Integer>> result , List<Integer> temp, int[] nums){
            result.add(new ArrayList<>(temp));
            if(index >= nums.length) return;

            for(int i=index ; i<nums.length; i++){
                if(i > index && nums[i] == nums[i-1]) continue;
                temp.add(nums[i]);
                findSubset(i+1,result,temp,nums);
                temp.remove(temp.size()-1);

            }
        }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        findSubset(0,result,temp,nums);
        return result;
    }
}